common/OpTestInstallUtil.py

```diff
@@ -0,0 +1,313 @@
+                print "failed to unmount", abs_repo_path
+                return ""
+        elif os.path.isdir(repo_path):
+            shutil.rmtree(repo_path)
+        else:
+            pass
+        if not os.path.isdir(repo_path):
+            os.makedirs(abs_repo_path)
+
+        if os.path.isfile(cdrom):
+            cdrom_path = cdrom
+        else:
+            cdrom_url = urllib2.urlopen(cdrom)
+            if not cdrom_url:
+                print "Unknown cdrom path %s" % cdrom
+                return ""
+            with open(os.path.join(BASE_PATH, "iso"), 'wb') as f:
+                f.write(cdrom_url.read())
+            cdrom_path = os.path.join(BASE_PATH, "iso")
+        cmd = "mount -t iso9660 -o loop %s %s" % (cdrom_path, abs_repo_path)
+        status, output = commands.getstatusoutput(cmd)
+        if status != 0:
+            print "Failed to mount iso %s on %s\n %s", (cdrom, abs_repo_path,
+                                                        output)
+            return ""
+        return abs_repo_path
+
+    def extract_install_files(self, repo_path):
+        """
+        extract the install file from given repo path
+
+        :params repo_path: os repo path either local or remote
+        """
+        vmlinux_src = os.path.join(repo_path, BOOTPATH, VMLINUX)
+        initrd_src = os.path.join(repo_path, BOOTPATH, INITRD)
+        vmlinux_dst = os.path.join(BASE_PATH, VMLINUX)
+        initrd_dst = os.path.join(BASE_PATH, INITRD)
+        # let us make sure, no old vmlinux, initrd
+        if os.path.isfile(vmlinux_dst):
+            os.remove(vmlinux_dst)
+        if os.path.isfile(initrd_dst):
+            os.remove(initrd_dst)
+
+        if os.path.isdir(repo_path):
+            try:
+                shutil.copyfile(vmlinux_src, vmlinux_dst)
+                shutil.copyfile(initrd_src, initrd_dst)
+            except Exception:
+                return False
+        else:
+            vmlinux_file = urllib2.urlopen(vmlinux_src)
+            initrd_file = urllib2.urlopen(initrd_src)
+            if not (vmlinux_file and initrd_file):
+                print "Unknown repo path %s, %s" % (vmlinux_src, initrd_src)
+                return False
+            try:
+                with open(vmlinux_dst, 'wb') as f:
+                    f.write(vmlinux_file.read())
+                with open(initrd_dst, 'wb') as f:
+                    f.write(initrd_file.read())
+            except Exception:
+                return False
+        return True
+
+    def set_bootable_disk(self, disk):
+        """
+        Sets the given disk as default bootable entry in petitboot
+        """
+        self.system.sys_set_bootdev_no_override()
+        self.system.host_console_unique_prompt()
+        # FIXME: wait till the device(disk) discovery in petitboot
+        time.sleep(60)
+        cmd = 'blkid %s-*' % disk
+        output = self.console.run_command(cmd)
+        uuid = output[0].split(':')[1].split('=')[1].replace("\"", "")
```

Comment:
```[console-pexpect]#blkid /dev/disk/by-id/ata-ST1000NM0033_00LY398_00LY393IBM_Z1W4S7NW-*
blkid /dev/disk/by-id/ata-ST1000NM0033_00LY398_00LY393IBM_Z1W4^M
S7NW-*
/dev/disk/by-id/ata-ST1000NM0033_00LY398_00LY393IBM_Z1W4S7NW-part2: UUID="7df158e4-c985-4fd9-9fa4-8eda85be39a1"
[console-pexpect]#echo $?
echo $?
0
# LAST COMMAND EXIT CODE 0 ('0\r\n')
ERROR (2588.209s)

======================================================================
ERROR [2588.209s]: runTest (testcases.InstallHostOS.InstallHostOS)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins_data/sath/pridhivi/satheesh_tests/op-test-framework/testcases/InstallHostOS.py", line 128, in runTest
    OpIU.set_bootable_disk(self.host.get_scratch_disk())
  File "/home/jenkins_data/sath/pridhivi/satheesh_tests/op-test-framework/common/OpTestInstallUtil.py", line 242, in set_bootable_disk
    uuid = output[0].split(':')[1].split('=')[1].replace("\"", "")
IndexError: list index out of range

----------------------------------------------------------------------
Ran 1 test in 2588.209s
```
Here we are failing with another console width problem.
can you add command "stty cols 300; stty rows 30;"

Reply:
@pridhiviraj sure, I will add it now, but it is good to have them addressed globally :-), lets think about it later...

Reply:
@sathnaga Sure, yeah, that should be addressed in state machine itself. for now you add it. I will fix it later with across tests.